lib/galaxy/config.py

```diff
@@ -921,6 +942,30 @@ def parse(string):
         return [parse(v) for v in allowed_origin_hostnames if v]
 
 
+def get_core_config_file():
+    return os.environ.get('GALAXY_CONFIG_FILE') or 'config/galaxy.yml'
```

Comment:
I would not assume a default here - if GALAXY_CONFIG_FILE isn't set maybe just not do the reloading. Is there a way to tell what file uwsgi loaded? If that is injected somehow I'd trust that also.

Reply:
So, if this env var is not set, then check what uwsgi loaded (I would think it's possible - will check). Otherwise don't reload. 
I assume uwsgi could load one of these: https://github.com/galaxyproject/galaxy/blob/dev/scripts/galaxy-main#L63

Reply:
I'm sure somehow a config file is found and used - but I want to track that and not try to refind it from inside the app - if that makes sense. Re-finding with slightly different logic would error prone - it'd work 85% of the time and we'd be reloading unreferenced configs the other 15% depending on how the app is loaded. With the galaxy-main script in particular - we may want to somehow update the config dict or environment to record what config file we used.

Reply:
Yes, I agree on refinding. 
So galaxy-main is where (and when) the config file is determined? (i.e., that's the source of truth about where the configuration values come from?)

Reply:
I think this is how to find the config file properly when coming from a wsgi server:
https://github.com/galaxyproject/galaxy/pull/921/files#diff-46f86ea0d82b84642e4218279ef61442R59

```
if global_conf and "__file__" in global_conf:
```

galaxy-main is a different app entry point that isn't used by webservers. Not sure if it sets this or not - but it probably should.

Reply:
Found it:  config file is found in a util/ function find_config_file() (surprise...) Furthermore, that method is called from BaseAppConfiguration - so the config file is already available as app.config.config_file. Thus, there's no need to re-find it - so there's no need for this additional function. PR forthcoming. Marking as resolved.